### DISCUSSION TITLE
How do I get pybricks to connect to my working usb bluetooth device?

### DISCUSSION THREAD

**Discussion by randome45189** [pybricks/support#1933] [Q&A]
How do I get pybricks to connect to my working bluetooth usb adaptor? I get this error message. The bluetooth adaptor is correctly installed, working and can connect to things like my phone. Yes I am using edge and yes I managed to enable the experimental web connection thingy hidden in the settings.

![image](https://github.com/user-attachments/assets/98529f6c-04bc-497a-9168-67d30886b2e7)

**Comment by BertLindeman** [comment]
Hello,

> The bluetooth adaptor is correctly installed, working and can connect to things like my phone.

This does not make sure that Bluetooth Low Energy (BLE for short) is supported by your adapter.

I assume you use Windows.

### How to Check for BLE Support on Windows

1. **Search for PowerShell**:
   - Click the **Start** button or press **`Win`** on your keyboard.
   - Type **"PowerShell"** in the search bar.
   - Click on **"Windows PowerShell"** when it appears in the search results.

2. **Run the Command**:
   - Copy and paste the following command into the PowerShell window and press **Enter**:
     ```powershell
     Get-PnpDevice -Class Bluetooth | Select-Object Name
     ```

3. **Check the List**:
   - Look through the output for **"Microsoft Bluetooth LE Enumerator"** or any entry containing **"LE"**.
   - If you see **"Bluetooth LE Enumerator"**, this confirms your adapter supports BLE.

4. **Next steps**:
  - Use the steps in [this discussion entry](https://github.com/orgs/pybricks/discussions/270)

Bert

**Comment by laurensvalk** [comment]
What operating system version are you using?

**Comment by randome45189** [comment]
Windows 7, 32 bit with edge (ew).
Bluetooth adaptor is a tp-link ub500 which supports Bluetooth low energy
and like I said works perfectly with my devices. It even picks up the Lego
hub I'm trying to use it with. Problem seems to be with pybricks and its
interaction with edge. Is there another way like a standalone .exe for the
pybricks application that I can I stall separately? Or has this all been
dumped on edge to sort out? This is very frustrating! I can't even get out
of the starting gates.

**Reply by laurensvalk** [reply]
Unfortunately Windows 7 does not support Bluetooth in the browser.

**Comment by BertLindeman** [comment]
To get started (a bit) maybe you can start with chrome on your phone or a tablet?

It **is** a bit small, but I just tried on my android phone and it is doable.

Bert

**Comment by randome45189** [comment]
So is there another way besides edge? Why is this pushed into a web browser
in the first place. I really do not like using edge at the best of times.
Is there something else I can use like a stand alone dev environment I can
install? Is there any detailed documentation on alternatives?

**Reply by dlech** [reply]
Windows 7 just doesn't have proper support for Bluetooth Low Energy, so no matter what kind of app it is, it won't work with Pybricks. Using https://github.com/pybricks/pybricksdev is an alternative to the browser-based app, but it doesn't work on Windows 7 either for the same reason.

**Comment by randome45189** [comment]
So there's no way to turn off this low energy thing and use the regular
Bluetooth stack? My computer is using mains, power isn't an issue. It's not
like I need to conserve power...! It's  not like I'm  using a phone or
tablet and need to preserve battery power. So what are the requirements? Is
there any documentation?

**Reply by dlech** [reply]
Pybricks only works with Bluetooth Low Energy currently.

Docs: https://pybricks.com/learn/getting-started/what-do-you-need/

**Comment by randome45189** [comment]
Why is it using low energy? Is this the only protocol the Lego hub
recognises? Can it not use the regular Bluetooth protocol?

**Reply by dlech** [reply]
Some hubs only have BLE and not Bluetooth Classic, so we made all of the hubs work the same.

**Comment by randome45189** [comment]
Also are you implementing pyBlueZ libraries or is this no longer being
developed?

**Reply by dlech** [reply]
We use Bleak for BLE in Python.

**Comment by randome45189** [comment]
I have windows server 2012r2 available. Will that work?

**Comment by randome45189** [comment]
Also, I think I have windows server 2019 somewhere. I could upgrade 2012r2
to that.?

**Reply by dlech** [reply]
It needs to be Windows 10 or newer, so server 2019 should work. Server 2012r2 is equivalent to Windows 8.1, so is not supported.
